src/Dumplie/Infrastructure/Doctrine/Metadata/DoctrineStorage.php

```diff
@@ -0,0 +1,238 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Dumplie\Infrastructure\Doctrine\Metadata;
```

Comment:
I think I would put it under
```
Dumplie\Infrastructure\Doctrine\Dbal\Metadata
```
namespace. I'm going to work on Doctrine ORM integration and it would be good to keep them separated.